code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,312 @@
+	alert_type = null
+
+/datum/status_effect/salve_regen/on_apply()
+	if(!isxeno(owner))
+		return FALSE
+	owner.balloon_alert(owner, "Regenerating")
+	return TRUE
+
+/datum/status_effect/salve_regen/on_remove()
+	owner.balloon_alert(owner, "No longer regenerating")
+	return ..()
+
+/datum/status_effect/salve_regen/tick()
+	/// Amount healed by this ability.
+	var/heal_amount = round(owner.maxHealth * 0.01) // 1% of the owner's maximum health
+	/// Remainder of the above, after brute has been healed. Applies the remainder to burns.
+	var/heal_remainder = max(0, heal_amount - owner.getBruteLoss())
+
+	owner.adjustBruteLoss(-heal_amount)
+	owner.adjustFireLoss(-heal_remainder, TRUE)
+	owner.adjust_sunder(-1)
+	new /obj/effect/temp_visual/healing(get_turf(owner))
+	return ..()
+
+// ***************************************
+// *********** Enhancement
+// ***************************************
+/particles/drone_enhancement
+	icon = 'icons/effects/particles/generic_particles.dmi'
+	icon_state = "fire"
+	color = "#9828CC"
+	width = 100
+	height = 100
+	count = 1000
+	spawning = 3
+	lifespan = 9
+	fade = 12
+	grow = 0.04
+	velocity = list(0, 0)
+	position = generator("circle", 16, 16, NORMAL_RAND)
+	drift = generator("vector", list(0, -0.15), list(0, 0.15))
+	gravity = list(0, 0.8)
+	scale = generator("vector", list(0.1, 0.1), list(0.6,0.6), NORMAL_RAND)
+	rotation = 0
+	spin = generator("num", 10, 20)
+
+/datum/status_effect/drone_enhancement
+	id = "drone enhancement"
+	duration = -1
+	tick_interval = 2 SECONDS
+	alert_type = null
+	/// Used to track the giver of this buff.
+	var/mob/living/carbon/xenomorph/buffing_target
+	/// Used to track the owner of this buff.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// Used for particles. Holds the particles instead of the mob. See particle_holder for documentation.
+	var/obj/effect/abstract/particle_holder/particle_holder
+	/// Used to determine whether there is an existing Essence Link or not. Also allows access to its vars.
+	var/datum/status_effect/stacking/essence_link/existing_link
+	/// The plasma cost per tick of this ability.
+	var/plasma_cost
+	/// Damage bonus given by this ability.
+	var/damage_addition
+	/// Speed bonus given by this ability.
+	var/speed_addition
+
+/datum/status_effect/drone_enhancement/on_creation(mob/living/new_owner, mob/living/carbon/new_target)
+	buff_owner = new_owner
+	buffing_target = new_target
+	existing_link = buffing_target.has_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+	return ..()
+
+/datum/status_effect/drone_enhancement/on_apply()
+	plasma_cost = round(buffing_target.xeno_caste.plasma_max * 0.1) // 10% of the buff giver's maximum plasma
+	damage_addition = round(buff_owner.xeno_caste.melee_damage * 0.04) // 4% of the user's base damage APPLIED AS A MULTIPLIER.
+	speed_addition = round(buff_owner.xeno_caste.speed * 0.05) // 5% of the user's base speed.
+	buffing_target.xeno_caste.plasma_gain = 0 // Denies natural plasma gain while this is active.
```

Comment:
You calculate vars here and then, dont use them? fix pls

Reply:
Also also useless comments

Reply:
The vars are used, though.
I removed the useless comments.

Reply:
Should be fixed in next commit.